supplier_server_mock/supplier_server/app.py

```diff
@@ -170,7 +170,26 @@ def booking():
 @app.route('/v1/booking/<booking_id>', methods=['DELETE'])
 @authorization_header
 def cancel_booking(booking_id):
-    return ('', 204)
+    try:
+        booked_at, product_id = utils.decode_booking_data(booking_id)
+    except binascii.Error:
+        raise exceptions.BadRequest(1004, 'Missing booking', 'Required argument \'booking_id\' was not found')
```

Comment:
Error message in the specification is different ("Booking with ID ID_NUMBER doesn't exist")